File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        mp1 = [-1] * 256
        mp2 = [-1] * 256
        n = len(s)
        for i in range(n):
            ch1 = s[i]
            ch2 = t[i]
            if (mp1[ord(ch1)] != -1 and mp1[ord(ch1)] != ord(ch2)) or (mp2[ord(ch2)] != -1 and mp2[ord(ch2)] != ord(ch1)):
                return False
            mp1[ord(ch1)] = ord(ch2)
            mp2[ord(ch2)] = ord(ch1)
        return True

File: LeetCode/0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> mp1, mp2;
        int n = s.length();        
        for(int i = 0; i < n; i++) {
            char ch1 = s[i];
            char ch2 = t[i];            
            if ((mp1.find(ch1) != mp1.end() && mp1[ch1] != ch2) || (mp2.find(ch2) != mp2.end() && mp2[ch2] != ch1)) {
                return false;
            }
            mp1[ch1] = ch2;
            mp2[ch2] = ch1;            
        }        
        return true;
    }
};

// class Solution {
// public:
//     bool isIsomorphic(string s, string t) {
//         vector<int> mp1(256, -1);
//         vector<int> mp2(256, -1);
//         int n = s.length();        
//         for(int i = 0; i<n; i++) {
//             char ch1 = s[i];
//             char ch2 = t[i];            
//             if ((mp1[ch1] != -1 && mp1[ch1] != ch2) || (mp2[ch2] != -1 && mp2[ch2] != ch1)) {
//                 return false;
//             }
//             mp1[ch1] = ch2;
//             mp2[ch2] = ch1;            
//         }        
//         return true;
//     }
// };